### DISCUSSION TITLE
How to ignore subdirectory's frontmatter.json?

### DISCUSSION THREAD

**Discussion by segln** [estruyf/vscode-front-matter#814] [Q&A] [answered]
FM currently detects the subdirectory's frontmatter.json file and complains that the frontMatter.content.pageFolders directory does not exist.

The parent folder is not the intended workspace.
How can I stop detecting subdirs and prevent these error messages?

**Comment by estruyf** [comment]
Hi @segln, what I'm assuming is that you are working in the following workspace structure:

```
├── project-folder-1
├── project-folder-2
├── project-folder-3
│   ├── .frontmatter
│   ├── frontmatter.json
│   ├── content
```

FM will detect the `frontmatter.json` file and its project folder location. It will use that location to retrieve the files and folders.

Could you let me know how you configured the paths for your page folders?

> How can I stop detecting subdirs and prevent these error messages?

What do you mean with this? Do you want to use FM in your workspace? If not, you can disable an extension per workspace.

**Reply by segln** [reply]
My workspace structure is like this:
```
root
├── .gitignore
├── ... files and dirs...
│
├── content
│   ├── frontmatter.json
│   ├── .frontmatter
│   │   └── database
│   │       ├── pinnedItemsDb.json
│   │       └── taxonomyDb.json
│   ├── .vscode
│   │   └── settings.json
│   ├── archive
│   │   ├── post1
```

I usually use VSCode on the root directory. I do not need FM for the root, just for the `content` directory.

However, every time I open the root directory, FM detects `content`'s `frontmatter.json`, populates `.frontmatter` in the root directory, and shows the error message: 

```
Front Matter CMS: Folder 'post1 ([[workspace]]/archive/post1)' does not exist. Please remove it from the settings.
```

Yes, `post1` is in the `content`, not in the root.
I do not need to detect the subdirectory's `frontmatter.json`. Is there an option to turn this off?

**Reply by estruyf** [reply, marked as answer]
The best option is to go to the extensions panel in the project and disable Front Matter CMS for the current workspace.

![image](https://github.com/estruyf/vscode-front-matter/assets/2900833/79186104-0e16-4459-8a16-e909c7477e51)

**Reply by segln** [reply]
I'll use it that way. Thank you for your help.
